File: kdcreader_android/android/src/main/kotlin/koamtac/kdc/sdk/plugins/KEvent.kt
package koamtac.kdc.sdk.plugins

class KEvent {
    object Name {
        const val CONNECTION_STATE_CHANGED = "onConnectionChanged"
        const val BARCODE_DATA_RECEIVED = "onBarcodeDataReceived"
        const val NFC_DATA_RECEIVED = "onNfcDataReceived"
        const val MSR_DATA_RECEIVED = "onMsrDataReceived"
        const val GENERAL_DATA_RECEIVED = "onDataReceived"
        const val POS_DATA_RECEIVED = "onPosDataReceived"
        const val ERROR_RECEIVED = "onErrorReceived"
        const val DEVICE_SCANNED = "onDeviceScanned"
        const val DEVICE_SCAN_FAILED = "onDeviceScanFailed"
        const val USB_DEVICE_ATTACHED = "onUsbDeviceAttached"
        const val USB_DEVICE_DETACHED = "onUsbDeviceDetached"

        const val SW_BARCODE_DATA_RECEIVED = "onSwBarcodeDataReceived"
    }
}

File: kdcreader_ios/ios/Classes/KEvent.swift
//
//  KEvent.swift
//  kdcreader_ios
//
//  Created by koamtac on 3/20/24.
//

import Foundation

struct KEvent {
    struct Name {
        static let CONNECTION_STATE_CHANGED = "onConnectionChanged"
        
        static let BARCODE_DATA_RECEIVED = "onBarcodeDataReceived"
        static let NFC_DATA_RECEIVED = "onNfcDataReceived"
        static let MSR_DATA_RECEIVED = "onMsrDataReceived"
        static let GENERAL_DATA_RECEIVED = "onDataReceived"
        
        static let POS_DATA_RECEIVED = "onPosDataReceived"
        
        static let ERROR_RECEIVED = "onErrorReceived"
        
        static let DEVICE_SCANNED = "onDeviceScanned"
        
        static let INFO_UPDATED = "onInfoUpdated"

        // Software Decoder
        static let SW_BARCODE_DATA_RECEIVED = "onSwBarcodeDataReceived"

        private init() {}
    }

    private init() {}
}
